packages/image_picker/image_picker_platform_interface/lib/src/types/media_selection_options.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../image_picker_platform_interface.dart';
+
+/// Specifies options class for selecting items when using [ImagePickerPlatform.getMedia].
+///
+/// This class inheritance is a byproduct of the api changing over time.
+/// It exists solely to avoid breaking changes.
+class MediaSelectionOptions extends ImageOptions {
+  /// Construct a new MediaSelectionOptions instance.
+  MediaSelectionOptions({
+    super.maxHeight,
+    super.maxWidth,
+    super.imageQuality,
+    super.requestFullMetadata,
```

Comment:
A MediaSelectionOptions should take an `ImageSelectionOptions` (or whatever the right one is) instead of all of its components.